core/templates/services/math-interactions.service.ts

```diff
@@ -171,9 +181,38 @@ export class MathInteractionsService {
       this.warningText = 'Your answer contains an invalid character: "_".';
       return false;
     }
-    if (expressionString.match(/(\+$)|(\+\))/g)) {
+    if (expressionString.match(/^(\+$)|^(\(\+\))/g)) {
```

Comment:
Can you explain all these regexes a bit more, with comments?

Also, is this not detectable from the error message produced by nerdamer? I'd imagine that would be more robust...

Reply:
Sure, So this regex is to give error message, `Your answer seems to be missing a number before the + operator.` when user writes input only as `+ or (+)` in the input field which is wrong.
Initially error message was `Your answer seems to be missing a variable/number after the "+."`

Reply:
I don't think I follow -- if the error message from nerdamer is `Your answer seems to be missing a variable/number after the "+"`, then that seems reasonable too, right?

I think my main concern with all these regexes are: how do you know they are foolproof and catch all cases (e.g. if the missing operator is in a parentheses etc.)? It would be bad if we merge this change and end up mistakenly declaring some answers as wrong when they are correct, or vice versa. So I was wondering if we could possibly get some additional hints from the nerdamer message etc.

Ditto elsewhere below.

Reply:
> I don't think I follow -- if the error message from nerdamer is `Your answer seems to be missing a variable/number after the "+"`, then that seems reasonable too, right?
> 
> I think my main concern with all these regexes are: how do you know they are foolproof and catch all cases (e.g. if the missing operator is in a parentheses etc.)? It would be bad if we merge this change and end up mistakenly declaring some answers as wrong when they are correct, or vice versa. So I was wondering if we could possibly get some additional hints from the nerdamer message etc.
> 
> Ditto elsewhere below.

Error message from nerdamer comes- `Unexpected token.` for `+ or (+)` operators so for them we can update the error message.
and for `- or (-) ` operators similarly some unidetificable error message was coming.
So I think we can change for these two operators using specific regex.

Reply:
> > I don't think I follow -- if the error message from nerdamer is `Your answer seems to be missing a variable/number after the "+"`, then that seems reasonable too, right?
> > I think my main concern with all these regexes are: how do you know they are foolproof and catch all cases (e.g. if the missing operator is in a parentheses etc.)? It would be bad if we merge this change and end up mistakenly declaring some answers as wrong when they are correct, or vice versa. So I was wondering if we could possibly get some additional hints from the nerdamer message etc.
> > Ditto elsewhere below.
> 
> Error message from nerdamer comes- `Unexpected token.` for `+ or (+)` operators so for them we can update the error message. and for `- or (-) ` operators similarly some unidetificable error message was coming. So I think we can change for these two operators using specific regex.
Also we are changing error message for these for cases only:
1. when user enter only `+ or (+)`
2. when user enter only `- or (-)` 
So it will not affect other cases and give nerdamer message for them.